tests/test_sample.py

```diff
@@ -0,0 +1,59 @@
+import os.path as op
+import cooler
+
+import cooltools
+import cooltools.api
+from numpy import testing
+
+
+def test_sample(request):
+    # perform test:
+    clr = cooler.Cooler(op.join(request.fspath.dirname, "data/CN.mm9.1000kb.cool"))
+
+    cooltools.api.sample.sample(
+        clr,
+        op.join(request.fspath.dirname, "data/CN.mm9.1000kb.test_sampled.cool"),
+        frac=0.5,
+    )
+    clr_result = cooler.Cooler(
+        op.join(request.fspath.dirname, "data/CN.mm9.1000kb.test_sampled.cool")
+    )
+    # Test that deviation from expected total is very small
+    testing.assert_allclose(clr_result.info["sum"], clr.info["sum"] / 2, rtol=1e-3)
+
+    cooltools.api.sample.sample(
+        clr,
+        op.join(request.fspath.dirname, "data/CN.mm9.1000kb.test_sampled.cool"),
+        count=200000000,
+    )
+    clr_result = cooler.Cooler(
+        op.join(request.fspath.dirname, "data/CN.mm9.1000kb.test_sampled.cool")
+    )
+    # Test that deviation from expected total is very small
+    testing.assert_allclose(clr_result.info["sum"], 200000000, rtol=1e-3)
+
+
+# Exact sampling is very slow! So commented out
+# cooltools.api.sample.sample(
+#     clr,
+#     op.join(request.fspath.dirname, "data/CN.mm9.1000kb.test_sampled.cool"),
```

Comment:
can we use a smaller input sample to uncomment this ?

Reply:
This is our normal test cooler at 1Mb resolution... So quite small! Idk we could add a 10Mb resolution one?

Reply:
Even with 10Mb resolution it takes a while... This whole file takes 95 sec to run through pytest on my laptop, which I guess is acceptable?

Reply:
let's try this way - w'd need to optimize out tests at some point anyways